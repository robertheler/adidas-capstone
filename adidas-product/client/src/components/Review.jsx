import React, { Component } from 'react';
import styled from 'styled-components';
import Stars from './Stars.jsx';

const RenderDiv = styled.div`
  display: flex;
  justify-content: space-between;
  font-weight: 700;
  font-style: bold;
  font-family: AdihausDIN,Helvetica;
  font-size: 11px;
  word-wrap: normal;
  margin: 0 0 20px 0;
`;

const StyledA = styled.span`

  text-decoration: underline;
`
const StyledCollection = styled.div`
  font-weight: 400;
  margin: 10 0 2px 0;
  color: #767677;
  word-wrap: normal;
  text-transform: uppercase;
  letter-spacing: 2px;
`;

const StyledSpan = styled.span`
color: black;
  &:hover {
    color: #fff; !important;
    background-color: #000; !important;
  }
  cursor: pointer;
`

class Review extends Component {
  constructor(props){
    super(props);
  }

  //TODO Render 5 stars filled appropriately
  render() {
    return (
      <RenderDiv>
        <StyledCollection>{this.props.product.collection_name}</StyledCollection>
        <StyledSpan>
          <Stars review={this.props.product.review_average}></Stars>
          <StyledA href="">{this.props.product.review_count}</StyledA>
        </StyledSpan>
      </RenderDiv>
    );
  }
}

export default Review;