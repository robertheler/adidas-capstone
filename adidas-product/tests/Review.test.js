import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Review from '../client/src/components/Review';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0],
    isSelected: false,
    handleColorChange: () => {}
  };
  const wrapper = shallow(<Review color={props.color} product={props.product} isSelected={props.isSelected} handleColorChange={props.handleColorChange}/>);
  return { wrapper, props };
}

describe('Review Component Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});