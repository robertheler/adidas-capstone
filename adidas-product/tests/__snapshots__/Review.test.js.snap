// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component Test Suite It should render correctly 1`] = `
<styled.div>
  <styled.div>
    Training
  </styled.div>
  <styled.span>
    <Stars
      review="3.52"
    />
    <styled.a
      href=""
    >
      394
    </styled.a>
  </styled.span>
</styled.div>
`;
